rfd/0194-spiffe-x509-override.md

```diff
@@ -0,0 +1,82 @@
+---
+authors: Edoardo Spadolini (edoardo.spadolini@goteleport.com)
+state: draft
+---
+
+# RFD 0194 - SPIFFE X.509 issuer override for external PKI support
+
+## Required Approvers
+
+* Engineering: @strideynet || @timothyb89
+
+## What
+
+Teleport manages a certificate authority to issue SPIFFE credentials, consisting of one or more keys used for JWT-SVID issuance and one or more keys and self-signed CA X.509 certificates for X509-SVID issuance. This RFD proposes adding support for overriding the X.509 certificates used as issuers for X509-SVID credentials.
+
+## Why
+
+There are a number of reasons why an organization may desire, or require, that Teleport Workload Identity credentials are issued by an intermediate to a root certificate authority that is controlled by the organization.
+
+For example:
+
+- It may be required as part of compliance policy.
+- Ability to revoke certificates issued by Teleport Workload Identity, without using Teleport Workload Identity.
+- Ability to rotate the Teleport Workload Identity certificate authority without needing to distribute the new certificate authority to validators working with workload identities.
+
+## Issuer override
+
+Currently, X509-SVID issuance in the Auth Service selects the first key usable by the Auth and signs a certificate for the client's public key using the X.509 certificate associated with the key as the issuer. We will introduce a new resource with kind `workload_identity_x509_issuer_override`:
+
+```yaml
+kind: workload_identity_x509_issuer_override
+version: v1
+metadata:
+  name: default
+spec:
+  overrides:
+    - issuer: base64 DER certificate
```

Comment:
I wonder if PEM wrapped may make more sense? Although doesn't feel like it's much different if it's base64 encoded anyway.

Reply:
The (hopefully not really used) UX of managing the override resource by editing yaml or json is a bit better with base64-encoded binary DER certs with no newlines compared to having to deal with newlines in yaml or escaping them in json (or, even worse, doing base64-encoded PEM).